app/src/main/kotlin/com/wire/android/ui/authentication/devices/remove/RemoveDeviceViewModel.kt

```diff
@@ -73,16 +66,57 @@ class RemoveDeviceViewModel @Inject constructor(
         updateStateIfDialogVisible { it.copy(error = RemoveDeviceError.None) }
     }
 
-    fun onItemClicked(device: Device) {
-        updateStateIfSuccess { it.copy(removeDeviceDialogState = RemoveDeviceDialogState.Visible(device = device)) }
+    fun onItemClicked(client: Client) {
+        tryToDeleteOrShowPasswordDialog(client)
+    }
+
+    private fun tryToDeleteOrShowPasswordDialog(client: Client) {
+        launchOnMain {
+            // try delete with no password (will success only for SSO accounts
```

Comment:
can't we store the type of the account when we do login? then we can ignore this kind of try errors?

Reply:
I'm trying to keep the flow [as simple as possible](https://en.wikipedia.org/wiki/KISS_principle).
and having multiple accounts type is something we don't have at the moment.